### ISSUE TITLE
runtime/trace, cmd/trace: provide a per-M view

### ISSUE THREAD

**Issue by prattmic** [golang/go#44074]
The Go execution tracer currently provides a per-P view (default, via `/trace`) or a per-G view (via `/goroutines`).

When investigating issues with the scheduler I've frequently found myself wanting a per-M view, as that is the level that the scheduler truly operates on. This makes it easier to see when Ps switch between Ms, when Ms stop, etc.

For #43997, I found the need compelling enough, that I built a one-off CL for it in http://golang.org/cl/286972. That CL adds an argument to (almost) every trace type, so I don't think it is acceptable as a general solution, but it would be nice to have a more complete version we could include for real.

cc @mknyszek @aclements @hyangah

**Comment by zhouguangyuan0718** [comment]
Sorry for bother.  Is this feature will be accepted in the future?
@prattmic

**Comment by prattmic** [comment]
> Is this feature will be accepted in the future?

I don't expect much pushback on this feature, however my CL https://golang.org/cl/286972 adds data to the trace file, thus potentially causing a performance impact, meaning it needs more thought, and no one is working on this. If my CL was free, then it likely would already have gone in.

**Comment by prattmic** [comment]
As a bit of an extension of this, I also dream of extending the trace files in per-M view for "runtime-internals view". I want a complete breakdown of what Ms are doing. i.e., from the moment an M wakes out of `mPark` until it sleeps again in `mPark` I want a fully specified that show what that M is doing: looking for work, running a goroutine, etc. Time in `usleep` or waiting on a `lock` may be blank, but should be annotated. `wakep`, `handoffp`, etc should flow events to the M they wake.
